cypress/fixtures/boundarylayer.json

```diff
@@ -0,0 +1,249 @@
+{
+  "totalSize": 503173,
+  "duplicates": 30,
+  "nonDeterministicResponses": 8,
+  "requests": [
+    {
+      "url": "http://localhost:8080/api/31/userSettings",
```

Comment:
@varl This is a great point, thanks for spotting it!

The default should be localhost:8080 in most development and staging environments, but we do want to enable overriding that.  This is currently possible, but very unwieldy:

The API url can be specified for the Cypress tests with the `CYPRESS_EXTERNAL_API` env var, and for the application itself (in Maps at least) with `DHIS2_HOME`.  So it is currently possible to point the application at `localhost:8080` in `DHIS2_HOME` and then do `yarn cy:start` to have it successfully stub the  network.  That's not good enough though.

I've made this somewhat easier in my most recent commit, so that the fixtures only store the path rather than the full URL.  The value in `CYPRESS_EXTERNAL_API` is prepended to the path to match full network requests when stubbing.  With this in place, you should be able to run against the committed fixtures with `CYPRESS_EXTERNAL_API=https://dhis2.vardevs.se/dev yarn cy:open`, assuming you have a running instance of maps on `localhost:8082` also pointing at `https://dhis2.vardevs.se/dev`.  Note that you can also override the URL of the running application-under-test with `CYPRESS_BASE_URL=http://localhost:8888` if needed.

The right solution to this will also involve a managed backend instance with stable starting database state, to prevent contextual divergence across testing environments.  This is currently possible with Docker/docker-compose/DHIS2-Backend but needs some more polish and performance improvements before it's a viable solution for regular use.

Does this address your point for now?  There are certainly improvements we can make along with the API configuration overhaul you outlined above.  Let me know if you have issues running locally!

Reply:
(Caveat to this - see below)

Reply:
Thanks, it works as a workaround for now, but it's too painful for regular and widespread use.

More importantly, I want to clarify something which concerns me:

The assumption that the default frontend developer's backend is on `localhost:8080` is incorrect.

A frontend developer should not _need_ to run the backend on the same machine at all, and we should not be building up our infrastructure nor developer environments on that assumption. This goes directly against my architectural vision. It affects the work related to Docker, tests, web application architecture, etc., so it is an important point. Please keep that in mind.

>I've just now made this somewhat easier in my most recent commit, so that the fixtures only store the path rather than the full URL. The value in CYPRESS_EXTERNAL_API is prepended to the path to match full network requests when stubbing.

Nice, this is a more flexible approach. 👍  If we can resolve the caveat about the Core sending back absolute URLs that also helps.